### DISCUSSION TITLE
How to generate @connection types for testing purpose with dgs codegen plugin

### DISCUSSION THREAD

**Discussion by ruxiao16** [Netflix/dgs-framework#1510] [Q&A]
Hello everyone,

I am working on a project with dgs codegen plugin. 
one of the type defined in the schema file is 
```
hello(first:Int, after:String): MessageConnection

movies(name: String): [Movie]
```
and 
```
type Message @connection {
    name: String
}

type Movie {
    name: String
    type: String
    director: Person
    actors(first: Int, after: String): PersonConnection
}

type Person @connection {
    name: String
    id: Int
}

```
and in my build.gradle i added type mapping as 
```
generateJava{
  ...

  typeMapping = ["PersonConnection": "graphql.relay.Connection<Person>",
                             "MessageConnection": "graphql.relay.SimpleListConnection<Message>"]
}
```

However,  when i try to write a test class for the defined endpoints i get below errors bean has failed to be :

`Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dgsQueryExecutor' defined in com.netflix.graphql.dgs.autoconfig.DgsAutoConfiguration: Unsatisfied dependency expressed through method 'dgsQueryExecutor' parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'schema' defined in com.netflix.graphql.dgs.autoconfig.DgsAutoConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'schema' threw exception; nested exception is SchemaProblem{errors=[The field type 'PersonConnection' is not present when resolving type 'Movie' [@40:1], The field type 'MessageConnection' is not present when resolving type 'Query' [@1:1], 'Person' [@47:1] tried to use an undeclared directive 'connection', 'Message' [@52:1] tried to use an undeclared directive 'connection']}
`

Test class looks like 
```
@SpringBootTest(classes = {DgsAutoConfiguration.class, foo.class, blah.class})
public class PurchaseProcessDataFetcherTest {

    @Autowired
    DgsQueryExecutor dgsQueryExecutor;
}
```

I feel like i am missing something in build.gradle file but i cannot figure out what it is. I have searched for like dgscode gen test configuration but to no avail.

**Comment by cccs-ian** [comment]
Does your project depend on `com.netflix.graphql.dgs:graphql-dgs-pagination`? This package contains the `@connection` directive definition your error seems to complain about

**Reply by ruxiao16** [reply]
yes it does. Turns out I have to add DgsPaginationTypeDefinitionRegistry inside the SpringBootTest classes.
